src/main/java/org/mskcc/cmo/metadb/service/impl/LabelGenMessageHandlingServiceImpl.java

```diff
@@ -177,7 +177,8 @@ public void run() {
                         // update contents of 'samples' in request json map to publish
                         // and add updated request json to publisher queue
                         Map<String, Object> requestJsonMap = mapper.readValue(requestJson, Map.class);
-                        requestJsonMap.put("samples", updatedSamples);
+                        requestJsonMap.put("samples",
+                                updatedSamples.toArray(new Object[updatedSamples.size()]));
```

Comment:
not all implementations of List allows modifications to the object - the List returned by Arrays.asList(..) is one such implementation so this change makes a copy but of the modifiable List implementation